src/components/MainNav/ProfileDropdown/ProfileDropdown.js

```diff
@@ -228,27 +242,40 @@ class ProfileDropdown extends Component {
 
   renderProfileTrigger = ({ getTriggerProps, open }) => {
     const { intl } = this.props;
-    const servicePointName = get(this.getUserData(), 'curServicePoint.name', null);
 
     return (
       <NavButton
         ariaLabel={intl.formatMessage({ id: 'stripes-core.mainnav.myProfileAriaLabel' })}
         selected={open}
         className={css.button}
         icon={this.getProfileImage()}
-        label={servicePointName ? (
-          <>
-            <span className={css.button__label}>
-              {servicePointName}
-            </span>
-            <Icon icon={open ? 'caret-up' : 'caret-down'} />
-          </>
-        ) : null}
+        label={this.renderProfileTriggerLabel({ open })}
         {...getTriggerProps()}
       />
     );
   }
 
+  renderProfileTriggerLabel = ({ open }) => {
+    const { okapi } = this.props.stripes;
+    const userData = this.getUserData();
+    const servicePointName = userData?.curServicePoint?.name;
+    const tenantName = userData?.tenants?.find(({ id }) => id === okapi.tenant)?.name;
+
+    const withLabel = Boolean(servicePointName || tenantName);
```

Comment:
A better convention than `with...` is `is...` or `has...`. We use `withFoo` as a naming convention for HOCs that provide a `Foo` object to the component they wrap.

Reply:
agree, missed this migrating from https://github.com/folio-org/stripes-core/pull/1307 PR